[pipelining] Test schedules against manual stage

kwen2501 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #126735
* #126721
* #126812

Added manual stage in test_schedule.py so that we can test various schedules against it.

In this file we now have:
- test_schedule_with_tracer
- test_schedule_with_manual
- test_grad_with_tracer
- test_grad_with_manual

Tested schedules are:
- ScheduleGPipe
- Schedule1F1B

cc @mrshenli @pritamdamania87 @zhaojuanmao @satgera @gqchen @aazzolini @osalpekar @jiayisuse @H-Huang @awgu @penguinwu @fegin @XilunWu @wanchaol @fduwjj @wz337 @tianyu-l @wconstab @yf225 @chauhang @d4l3k

kwen2501 committed:
[pipelining] Test schedules against manual stage

[ghstack-poisoned]

wconstab commented on test/distributed/pipelining/test_schedule.py:
```diff
@@ -100,8 +101,51 @@ def test_ec_backward(self, ScheduleClass):
     @parametrize("ScheduleClass", [ScheduleGPipe, Schedule1F1B])
-    def test_grad(self, ScheduleClass):
-        mod = MultiMLP(d_hid)
+    def test_schedule_with_manual(self, ScheduleClass):
```
just curious- is there anything that test_schedule* tests that test_grad* does not test? or should we just delete test_schedule and rename test_grad to test_schedule?

kwen2501 replied:
Makes sense, maybe I can consolidate them.

H-Huang reviewed (approved):
LGTM!

kwen2501 committed:
Update on "[pipelining] Test schedules against manual stage"


Added manual stage in test_schedule.py so that we can test various schedules against it.

In this file we now have:
- test_schedule_with_tracer
- test_schedule_with_manual
- test_grad_with_tracer
- test_grad_with_manual

Tested schedules are:
- ScheduleGPipe
- Schedule1F1B

cc mrshenli pritamdamania87 zhaojuanmao satgera gqchen aazzolini osalpekar jiayisuse H-Huang awgu penguinwu fegin XilunWu wanchaol fduwjj wz337 tianyu-l wconstab yf225 chauhang d4l3k

[ghstack-poisoned]

kwen2501 committed:
Update on "[pipelining] Test schedules against manual stage"


Added manual stage in test_schedule.py so that we can test various schedules against it.

In this file we now have:
- test_schedule_with_tracer
- test_schedule_with_manual
- test_grad_with_tracer
- test_grad_with_manual

Tested schedules are:
- ScheduleGPipe
- Schedule1F1B

cc mrshenli pritamdamania87 zhaojuanmao satgera gqchen aazzolini osalpekar jiayisuse H-Huang awgu penguinwu fegin XilunWu wanchaol fduwjj wz337 tianyu-l wconstab yf225 chauhang d4l3k

[ghstack-poisoned]

kwen2501 committed:
Update on "[pipelining] Test schedules against manual stage"


Added manual stage in test_schedule.py so that we can test various schedules against it.

In this file we now have:
- test_schedule_with_tracer
- test_schedule_with_manual
- test_grad_with_tracer
- test_grad_with_manual

Tested schedules are:
- ScheduleGPipe
- Schedule1F1B

cc mrshenli pritamdamania87 zhaojuanmao satgera gqchen aazzolini osalpekar jiayisuse H-Huang awgu penguinwu fegin XilunWu wanchaol fduwjj wz337 tianyu-l wconstab yf225 chauhang d4l3k

[ghstack-poisoned]

wconstab commented:
@pytorchbot merge
